script.speedtester/testit.py

```diff
@@ -0,0 +1,1070 @@
+#^.___~"--._	~-{	 .-~ .	`\ Y . /	|								   #
+# <__ ~"-.	~		/_/	  \	  \I  Y	  : |								   #
+#	^-.__			~(_/   \   >._:	  | l______							   #
+#		^--.,___.-~"  /_/	!  `-.~"--l_ /	   ~"-.						   #
+#			   (_/ .  ~(   /'	  "~"--,Y	-=b-. _)					   #
+#				(_/ .  \	Dr0idGuy  / l	   c"~o \					   #
+#				 \ /	`.	  .		.^	 \_.-~"~--.	 )					   #
+#				  (_/ .	  `	 /	   /	   !	   )/					   #
+#				   / / _.	'.	 .':	  /		   '					   #
+#				   ~(_/ .	/	 _	`  .-<_								   #
+#					 /_/ . ' .-~" `.  / \  \		  ,z=.				   #
+#					 ~( /	'  :   | K	 "-.~-.______//					   #
+#					   "-,.	   l   I/ \_	__{--->._(==.				   #
+#						//(		\  <	~"~"	 //						   #
+#					   /' /\	 \	\	  ,v=.	((						   #
+#					 .^. / /\	  "	 }__ //===-	 `						   #
+#					/ / ' '	 "-.,__ {---(==-							   #
+#				  .^ '		 :	T  ~"	ll								   #
+#				 / .  .	 . : | :!		 \								   #
+#				(_/	 /	 | | j-"		  ~^							   #
+#				  ~-<_(_.^-~"											   #
+#																		   #
+#																		   #
+#					   original speedtest-cli DEV						   #
+#			 Matt Martz (https://github.com/sivel/speedtest-cli)		   #
+#																		   #
+#																		   #
+# Licensed under the Apache License, Version 2.0 (the "License"); you may  #
+# not use this file except in compliance with the License. You may obtain  #
+# a copy of the License at												   #
+#																		   #
+#	   http://www.apache.org/licenses/LICENSE-2.0						   #
+#																		   #
+# Unless required by applicable law or agreed to in writing, software	   #
+# distributed under the License is distributed on an "AS IS" BASIS,WITHOUT #
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the #
+# License for the specific language governing permissions and limitations  #
+# under the License.													   #
+############################################################################
+
+
+try:
+	from kodi_six import xbmc, xbmcaddon, xbmcgui
+except ImportError:
+	import xbmc ,xbmcgui, xbmcaddon
+
+import os
+import re
+import sys
+import math
+import signal
+import socket
+import timeit
+import platform
+import threading
+import time
+import uuid
+
+
+try:
+	import xml.etree.cElementTree as ET
+	from xml.dom import minidom as DOM
+except ImportError:
+	try:
+		import xml.etree.ElementTree as ET
+	except ImportError:
+		from xml.dom import minidom as DOM
+		ET = None
+
+ADDON = xbmcaddon.Addon()
+ADDONID = ADDON.getAddonInfo('id')
+ADDONNAME = ADDON.getAddonInfo('name')
+ADDONVERSION = ADDON.getAddonInfo('version')
+ART = xbmc.translatePath(os.path.join(ADDON.getAddonInfo('path'),'resources', 'skins', 'Default', 'media'))
+
+source = None
+shutdown_event = None
+socket_socket = socket.socket
+ua_tuple = ('Mozilla/5.0', '(%s; U; %s; en-us)'
+		% (platform.system(), platform.architecture()[0]),
+		'Python/%s' % platform.python_version(),
+		'(KHTML, like Gecko)')
+		
+user_agent = ' '.join(ua_tuple)
+
+#py2 imports except py3 ver
```

Comment:
It's better and more maintainable to use `sys.version_info.major` instead of blindingly try except on the imports